pkgs/tools/admin/salt/default.nix

```diff
@@ -1,21 +1,43 @@
-{
-  stdenv, pythonPackages, openssl,
-
-  # Many Salt modules require various Python modules to be installed,
+{ lib
+, python3
+, openssl
+, # Many Salt modules require various Python modules to be installed,
   # passing them in this array enables Salt to find them.
   extraInputs ? []
 }:
+let
+
+  defaultOverrides = [
+    # can be unpinned once https://github.com/saltstack/salt/issues/56007 is resolved
+    (mkOverride "msgpack" "0.6.2"
+      "0c0q3vx0x137567msgs5dnizghnr059qi5kfqigxbz26jf2jyg7a")
+  ];
+
+  mkOverride = attrname: version: sha256:
+    self: super: {
+      ${attrname} = super.${attrname}.overridePythonAttrs (oldAttrs: {
+        inherit version;
+        src = oldAttrs.src.override {
+          inherit version sha256;
+        };
+      });
+    };
+
+  py = python3.override {
+    # Put packageOverrides at the start so they are applied after defaultOverrides
+    packageOverrides = lib.foldr lib.composeExtensions (self: super: { }) defaultOverrides;
+  };
 
-pythonPackages.buildPythonApplication rec {
+in with py.pkgs; buildPythonApplication rec {
```

Comment:
nit: I personally prefer to keep `with`s scoped as small as possible, e.g. to only use it for the `propagatedBuildInputs` and explicitly use `py.pkgs.{buildPythonApplication,fetchPypi}` as done previously